### DISCUSSION TITLE
Using semantic-release in a project with Node version < 18

### DISCUSSION THREAD

**Discussion by kapowaz** [semantic-release/semantic-release#2694] [Configuration] [answered]
The [Node version requirement][1] document points out that whilst semantic-release requires Node v18.0.0+, it can be used in CI ‘from a CI job running on the latest available LTS version of node’. In my case, I’m setting up semantic-release to run on my projects in Circle CI, and running this command as part of my release job ought to do the job:

```
$ nvm install 'lts/*' && npx semantic-release
```

Unfortunately it seems I can’t even have `semantic-release` in the package.json file for my project unless the project itself is also using Node 18; if I try to run `yarn install` with my Node version as (for example) v16.19.0, I get this error:

```
error semantic-release@20.1.0: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.19.0"
```

How can you reconcile the need for a different (lower) version of Node with the _package_ you’re using semantic-release to release, and the version of Node semantic-release needs _itself_? I’d appreciate any advice on how to resolve this.

[1]: https://github.com/semantic-release/semantic-release/blob/HEAD/docs/support/node-version.md

**Comment by travi** [comment, marked as answer]
since semantic-release is only needed for your release process, we usually recommend installing at the last possible moment before your release happens. `npx` is great for this since it can install as part of the execution process. that way you don't have it defined as a dependency in your `package.json` at all.

**Reply by kapowaz** [reply]
Thanks for the reply! I’m assuming that if I’m using any semantic-release plugins I can just leave these as dev dependencies, even though they report as having a missing peer dependency in semantic-release itself? I had been using `@semantic-release/changelog` and `@semantic-release/git` in my project. Is there a better way of doing this that doesn’t involve cluttering up my circle config.yml with an increasingly lengthy build step command? I’m envisaging something like:

```
nvm install 'lts/*' && yarn add -D @semantic-release/changelog @semantic-release/git npx semantic-release || true
```

**Reply by akarsh-kr** [reply]
@travi could you please help me to resolve this issue 
<img width="1349" alt="Image Pasted at 2023-7-28 10-51" src="https://github.com/semantic-release/semantic-release/assets/127098772/eda1a548-1e79-47ec-9ee7-c0498158c4b9">

**Reply by akarsh-kr** [reply]
It is not allowed to update node

**Reply by seebeen** [reply]
If you cannot update node to 18.x you need to downgrade semantic-release

**Reply by akarsh-kr** [reply]
@seebeen I am not able to find that on package.json, is there any particular way to downgrade it?

**Reply by seebeen** [reply]
If you're using it via npx then `npx semantic-release@19`

**Reply by akarsh-kr** [reply]
How can we ignore that while adding node_modules?

**Reply by travi** [reply]
> It is not allowed to update node

i would argue that this is the problem that you need to solve. node v16 will be [EOL on 2023-09-11](https://nodejs.dev/en/about/releases/). you need to be preparing for that date anyway, so we are past the point of providing specialized support for avoiding using a current version if you choose to use an older version.

the accepted answer for this discussion already provides our guidance for avoiding installation as a local dependency and configuring your ci pipeline to use a supported version for the release step. if you choose to avoid the recommended approach, you will need to sort through the details yourself.

**Comment by 88pktalha** [comment]
https://github.com/semantic-release/semantic-release/discussions/2585#discussion-4441345

**Reply by 88pktalha** [reply]
Tank's to all

**Comment by 88pktalha** [comment]
Abu Talha
